test/libyul/objectCompiler/eof/subObject.yul

```diff
@@ -0,0 +1,29 @@
+object "a" {
+  code {}
+  // Unreferenced data is not added to the assembled bytecode.
+  data "str" "Hello, World!"
```

Comment:
Looking at `Bytecode` below, this is clearly not true on EOF right now, so this needs a TODO (and will need to be implemented eventually).

Same in `data.yul` and `subSubObject.yul.`

Reply:
Fixed according to comment https://github.com/ethereum/solidity/pull/15662#discussion_r1937432410